### PR TITLE
add module names to vimdoc to avoid duplicate tags

### LINKED ISSUE #208: [Bug]: Duplicate tags `M.autocmds` and `M.whichkey_listen`
### Similar Issues

- [X] Before filing, I have searched for similar issues.

### Neovim Version

```
NVIM v0.8.0
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by nixbld

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "
/nix/store/217n2mm1g00fiss305mahpd9ksl2b812-neovim-unwrapped-0.8.0/share/nvim
"

Run :checkhealth for more info
```

### Expected Behavior

There should be only one `M.autocmds` and one `M.whickey_listen`

### Actual Behavior

https://github.com/mrjones2014/legendary.nvim/blob/67acc7d5ce7598ea159e1c689cc0f323bcbfb297/doc/legendary-api.txt#L32
https://github.com/mrjones2014/legendary.nvim/blob/67acc7d5ce7598ea159e1c689cc0f323bcbfb297/doc/legendary-api.txt#L120

https://github.com/mrjones2014/legendary.nvim/blob/67acc7d5ce7598ea159e1c689cc0f323bcbfb297/doc/legendary-api.txt#L83
https://github.com/mrjones2014/legendary.nvim/blob/67acc7d5ce7598ea159e1c689cc0f323bcbfb297/doc/legendary-api.txt#L347

### Minimal Configuration to Reproduce

```sh
nvim --headless "+helptags doc" +quit!
```

### Additional Details and/or Screenshots

_No response_

### PR DESCRIPTION
**Pull request by figsoda**
fixes #208

## How to Test

1. Please leave detailed testing instructions

## Testing for Regressions

I have tested the following:

- [ ] Triggering keymaps from `legendary.nvim` in all modes (normal, insert, visual)
- [ ] Creating keymaps via `legendary.nvim`, then triggering via the keymap in all modes (normal, insert, visual)
- [ ] Triggering commands from `legendary.nvim` in all modes (normal, insert, visual)
- [ ] Creating commands via `legendary.nvim`, then running the command manually from the command line
- [ ] `augroup`/`autocmd`s created through `legendary.nvim` work correctly

### CODE REVIEW

File: gen-api-docs.bash [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-lemmy-help \
+lemmy-help -fact \
```
Comment (mrjones2014) [inline gen-api-docs.bash:3]: What does `-fact` do?

**Review by mrjones2014 — COMMENTED** [review]
Thanks for the PR! Just left one question.

File: gen-api-docs.bash (hunk 1 above)
Comment (figsoda) [reply gen-api-docs.bash:3]: -f, --prefix-func       Prefix function name with ---@mod name
    -a, --prefix-alias      Prefix ---@alias tag with return/---@mod name
    -c, --prefix-class      Prefix ---@class tag with return/---@mod name
    -t, --prefix-type       Prefix ---@type tag with ---@mod name

I don't think all of them makes a difference atm, but I added all of them just to be future-proof

Comment (mrjones2014) [reply gen-api-docs.bash:3]: Got it. That makes sense. Thanks.

**Comment by mrjones2014** [comment]
Included this in a new tag/version: `v2.1.1`

**Pull request merged**